addons/sale_order_dates/models/sale_order.py

```diff
@@ -24,30 +24,28 @@ class SaleOrder(models.Model):
                                           "processed as soon as possible. In that case the expected "
                                           "date will be computed using the default method: based on "
                                           "the Product Lead Times and the Company's Security Delay.")
-    effective_date = fields.Date(compute='_compute_picking_ids', string='Effective Date', store=True,
-                                 help="Date on which the first Delivery Order was created.")
+    effective_date = fields.Date(compute='_compute_effective_date', string='Effective Date', store=True,
+                                 help="Date on which the first shipment was successfully delivered to customer.")
 
-    @api.depends('date_order', 'order_line.customer_lead')
+    @api.depends('order_line.customer_lead', 'confirmation_date', 'picking_policy')
```

Comment:
I think we could add some triggers, like order_line.state, maybe create_date if we use it (not sure).

Reply:
We can set trigger order_line.state to set  commitment date false when state goes to cancel,
and we don't need create_date on trigger.